src/js/heuristicblocking.js

```diff
@@ -125,12 +127,115 @@ HeuristicBlocker.prototype = {
       return {};
     }
 
-    // ignore if there are no tracking cookies
-    if (!hasCookieTracking(details, origin)) {
+    // check if there are tracking cookies
+    if (hasCookieTracking(details, origin)) {
+      this._recordPrevalence(fqdn, origin, tabOrigin);
       return {};
     }
 
-    this._recordPrevalence(fqdn, origin, tabOrigin);
+    let self = this;
+
+    // check if this request is a cookie-syncing pixel
+    chrome.cookies.getAll({}, function(cookies) {
+      self.pixelCookieSyncAccounting(details, cookies);
+    });
+  },
+
+  /**
+   * Checks for cookie syncing: requests to third-party domains that include
+   * high entropy data from first-party cookies.
+   *
+   * TODO: is it OK to call this from heuristicBlockingAccounting?
```

Comment:
No need for this TODO.

Reply:
OK yeah, the TODO should go, but great question ... `heuristicBlockingAccounting` is called by two separate async `chrome.webRequest` listeners, and sometimes by a sync (!) `webRequest` listener. That seems like too many callers for cookie sharing accounting.

We might want to add a new, dedicated async listener (at what stage in the request lifecycle?), filtered specifically to "image" resources.

Reply:
Or it might be simpler to just place further restrictions on when `pixelCookieSyncAccounting` is triggered inside `heuristicBlockingAccounting`. Only proceed for images, yes, but also only when called by `onBeforeSendHeaders` (if that's indeed the right stage in the request lifecycle for checking up on cookie sharing).